File: chapter-1/ruby/1.2.rb
require 'test/unit/assertions'
include Test::Unit::Assertions

# Write code to reverse a C-Style String. 
# (C-String means that “abcd” is represented as  ve characters, including the null character.)

# Complexity = O(N)
def reverse_c_style(str)
  if str.class != String then return end

  output = []
  i = str.size - 3

  while i >= 0
    output.push str[i] 
    i = i - 1
  end

  output.join('').concat('\0')
end

assert_equal 'fedcba\0', reverse_c_style('abcdef\0')

File: chapter-1/js/1.2.js
/*
Write code to reverse a C-Style String. 
(C-String means that “abcd” is represented as a five characters, including the null character.)
*/

function reverse(str) {
  if (typeof(str) != 'string') { return; }

  let output = '';
  for (let i=str.length-2; i >= 0; i--) {
    output = output + str[i];
  }

  return output+'\0';
};

var assert = require('assert');
assert.equal('fedcba\0', reverse('abcdef\0'));
